warp10/src/main/java/io/warp10/script/functions/REMOVE.java

```diff
@@ -67,8 +69,33 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
       
       stack.push(coll);
       stack.push(o);
+    } else if (coll instanceof GeoTimeSerie) {
+      if (!(key instanceof Long) && !(key instanceof Integer)) {
+        throw new WarpScriptException(getName() + " expects an integer as key.");
+      }
+      int idx = ((Number) key).intValue();
+
+      GeoTimeSerie gts = (GeoTimeSerie) coll;
+
+      int n = GTSHelper.nvalues(gts);
+
+      if (idx < 0) {
+        idx += n;
+      }
+
+      GeoTimeSerie pruned = gts.cloneEmpty(n);
+
+      for (int i = 0; i < n; i++) {
+        if (i == idx) {
+          continue;
+        }
+
+        GTSHelper.setValue(pruned, GTSHelper.tickAtIndex(gts, i), GTSHelper.locationAtIndex(gts, i), GTSHelper.elevationAtIndex(gts, i), GTSHelper.valueAtIndex(gts, i), false);
+      }
+
+      stack.push(pruned);
```

Comment:
Shouldn't REMOVE behave similarly on a GTS as it does on a list or map? i.e. leave on the stack the data point tuple it just removed? With [ 0 NaN NaN NaN NULL ] if the datapoint did not exist?

Reply:
You're right, fixing this.